fetchai/ledger/api/tx.py

```diff
@@ -38,6 +42,74 @@ def digest_bytes(self):
         return self._digest_bytes
 
 
+class TxContents:
+    def __init__(self,
+                 digest: bytes,
+                 action: str,
+                 chain_code: str,
+                 from_address: str,
+                 contract_digest: str,
+                 contract_address: str,
+                 valid_from: int,
+                 valid_until: int,
+                 charge: int,
+                 charge_limit: int,
+                 transfers: list,
+                 signatories: list,
+                 data: str
+                 ):
+        self._digest_bytes = digest
+        self._digest_hex = self._digest_bytes.hex()
+        self.action = action
+        self.chain_code = chain_code
+        self.from_address = from_address
+        self.contract_digest = contract_digest
+        self.contract_address = contract_address
+        self.valid_from = valid_from
+        self.valid_until = valid_until
+        self.charge = charge
+        self.charge_limit = charge_limit
+        self._transfers = transfers
+        self.signatories = signatories
+        self.data = data
+
+    def transfers_to(self, address):
+        total = 0
+        for item in self.transfers:
+            if item['to'] == address:
+                total += item['amount']
+
+    @property
+    def transfers(self):
+        result = defaultdict(list)
+        for item in self._transfers:
```

Comment:
Do I understand correctly that these are all the transfers of FET resulting from the transaction hitting the address (i.e. if it is a non-contract address there is exactly one item in the list, otherwise as many as there are transfers initiated by the contract call - this is assuming you distinguish between contract and non-contract addresses which I am not sure you are)?

Reply:
If you do separate between non-contract and contract addresses then for non-contract addresses you could expose the `amount` directly.

Reply:
I've simplified the interface here. Transfers are now stored as a dictionary of `{Address: amount}`, and the class has a single function `transfer_to`, which will return the amount of FET transfered to an address (or 0 if that address isn't part of the transaction)